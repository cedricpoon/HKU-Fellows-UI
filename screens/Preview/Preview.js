import React, { Component } from 'react';
import { Container, Content } from 'native-base';

import { PostScrollable, PreviewFooter, PreviewHeader } from 'hkufui/components';
import postList from 'hkufui/docs/postList.json';

class Preview extends Component {

  render() {
    return (
      <Container>
        <PreviewHeader location={postList.location} />
        <Content>
          <PostScrollable posts={postList.posts} />
        </Content>
        <PreviewFooter />
      </Container>
    );
  }
}

export default Preview;
