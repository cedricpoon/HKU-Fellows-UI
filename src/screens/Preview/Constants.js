export const SCROLLABLE_END_REACH_THRESHOLD = 0.35;

export const ITEM_HEIGHT = 82.5; /* predefined for better performance */
