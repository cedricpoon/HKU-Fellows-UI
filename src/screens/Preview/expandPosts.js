import { EXPANDING_POSTS, EXPANDED_POSTS, LAST_POST } from 'hkufui/src/constants/actionTypes';
import * as status from 'hkufui/src/constants/expandStatus';
import { login } from 'hkufui/config/webapi';

import { onLogin, onClear } from '../Login/authenticate';
import { onFill, retrievePosts } from './PostPreviewLoader/loadPosts';

async function invokeRetrieving(dispatch, getState) {
  const { credential, posts } = getState();

  try {
    const res = await retrievePosts(dispatch, getState);
    if (res.status === 200 || res.status === 204) {
      // successful return
      if (res.payload.length > 0) {
        dispatch(onFill(posts.posts.concat(res.payload)));
        dispatch(onExpanded());
      } else {
        dispatch(onLast());
      }
    } else {
      // failure
      switch(res.status) {
        case 408:
          // refresh token
          await dispatch(onLogin({
            credential: { username: credential.userId, passphrase: credential.passphrase },
            path: login.passphrase
          }));
          break;
        case 401:
          dispatch(onClear());
          break;
      }
      throw new Error();
    }
  } catch (error) {
    // try retrieve posts again
    invokeRetrieving(dispatch, getState);
  }
}

export function fetchExpansion() {
  return (dispatch, getState) => {
    dispatch(onExpanding());
    invokeRetrieving(dispatch, getState);
  };
}

export const onExpanding = () => ({
  type: EXPANDING_POSTS
});

export const onExpanded = () => ({
  type: EXPANDED_POSTS
});

export const onLast = () => ({
  type: LAST_POST
});

const handleExpandPosts = (state = {}, action = {}) => {
  switch (action.type) {
    case EXPANDING_POSTS:
      return {
        ...state,
        index: state.index + 1,
        subStatus: status.EXPANDING
      }
    case EXPANDED_POSTS:
      return {
        ...state,
        subStatus: status.BLAND
      }
    case LAST_POST:
      return {
        ...state,
        subStatus: status.HALT
      }
    default:
      return state;
  }
};

export default handleExpandPosts;
