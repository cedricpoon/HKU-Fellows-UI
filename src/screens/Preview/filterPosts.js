import { UPDATE_FILTER, UPDATE_QUERY, UPDATE_KEYWORD } from 'hkufui/src/constants/actionTypes';
import defaultState from 'hkufui/src/store/globalState';

export const onUpdateQuery = (query) => ({
  type: UPDATE_QUERY,
  payload: { query }
});

export const onUpdateKeyword = (keyword) => ({
  type: UPDATE_KEYWORD,
  payload: { keyword }
});

export function resetQueryKeyword() {
  return (dispatch) => {
    dispatch(onUpdateQuery(null));
    dispatch(onUpdateKeyword(null));
  };
}

export const onUpdateFilter = (filter) => ({
  type: UPDATE_FILTER,
  payload: { filter }
});

export const onResetFilter = () => ({
  type: UPDATE_FILTER,
  payload: { filter: defaultState.posts.filter }
});

const handleFilterPosts = (state = {}, action = {}) => {
  switch (action.type) {
    case UPDATE_QUERY:
      return {
        ...state,
        query: action.payload.query,
        keyword: null
      };
    case UPDATE_KEYWORD:
      return {
        ...state,
        query: null,
        hashtag: action.payload.keyword
      };
    case UPDATE_FILTER:
      return {
        ...state,
        filter: action.payload.filter
      };
    default:
      return state;
  }
};

export default handleFilterPosts;
