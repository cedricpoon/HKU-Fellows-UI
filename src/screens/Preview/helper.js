export function classifyQuery (query) {
  // this method should return either keyword object or solid query string
  if (query && /^#.*[^# ]$/.test(query)) {
    // keyword pattern: '#xxx#yyy'
    const keywords = query.replace(/ /g, '').split('#').filter(Boolean);
    return {
      primary: keywords[0],
      ...(keywords[1] && { secondary: keywords[1] })
    }
  } else {
    // query pattern: 'xxxyyy'
    return query;
  }
}
