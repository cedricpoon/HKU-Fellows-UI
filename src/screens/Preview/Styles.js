import { StyleSheet } from 'react-native';
import { fontSize, gridBase } from 'hkufui/theme/grid';
import { ITEM_HEIGHT } from './Constants';

export default StyleSheet.create({
  copyright: {
    marginTop: gridBase * 2,
    marginLeft: gridBase * 6,
    marginBottom: gridBase * 8,
    fontSize: fontSize.miniSize
  },
  heightHolder: {
    height: ITEM_HEIGHT
  }
});
