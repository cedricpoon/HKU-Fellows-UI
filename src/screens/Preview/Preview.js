import React, { Component } from 'react';
import { RefreshControl } from 'react-native';
import { Container, Text } from 'native-base';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { PreviewFooter, PreviewHeader, PostLoadIndicator } from 'hkufui/components';
import PostPreviewLoader from './PostPreviewLoader/PostPreviewLoader'

import * as _expandStatus from 'hkufui/src/constants/expandStatus';
import { BLAND, EXPANDING } from 'hkufui/src/constants/expandStatus';
import * as _loadStatus from 'hkufui/src/constants/loadStatus';
import { OK } from 'hkufui/src/constants/loadStatus';

import { fetchPostsSafe } from './PostPreviewLoader/loadPosts';
import { fetchExpansion } from './expandPosts';
import { onUpdateFilter, onUpdateQuery, onUpdateKeyword, resetQueryKeyword } from './filterPosts';
import styles from './Styles';
import { classifyQuery } from './helper';

import { circularTint } from 'hkufui/theme/palette';
import { localize } from 'hkufui/locale';
import { SCROLLABLE_END_REACH_THRESHOLD, ITEM_HEIGHT } from './Constants';

const locale = localize({ language: 'en', country: 'hk' });

export class Preview extends Component {
  constructor(props) {
    super(props);
    this.state = { willRefresh: false, isReady: false, searched: false };
  }

  _refresh = () => {
    this.props.onLoadPost();
  }

  _scrollInit = () => {
    this.setState({ isReady: true });
  }

  _loadMore = () => {
    if (this.state.isReady && this.props.expandStatus === BLAND) {
      this.props.onLoadMore();
    }
  }

  _scrollableUnmount = () => {
    this.setState({ isReady: false });
  }

  _renderRefreshControl = () => {
    return(
      <RefreshControl
        refreshing={false}
        onRefresh={this._refresh}
        tintColor={circularTint}
      />
    );
  }

  _renderListFooter = () => {
    const { expandStatus } = this.props;

    if (expandStatus === EXPANDING) {
      return (<PostLoadIndicator stalled />);
    }
    /* no more post footer */
    return (
      <Text note style={styles.copyright}>
        {locale['post.copyright']}
      </Text>
    );
  }

  _filterWrapper = (filter) => {
    const { onUpdateFilter } = this.props;
    return () => {
      onUpdateFilter(filter);
      this.footer.filterToggle();
    };
  }

  _searchWrapper = (query) => {
    const { onUpdateQuery, onUpdateKeyword } = this.props;
    return () => {
      const target = classifyQuery(query);
      this.setState({ searched: true });

      if (typeof target === 'object') {
        onUpdateKeyword(target);
      } else {
        onUpdateQuery(target);
      }
    }
  }

  componentDidUpdate() {
    const { shouldSearchBarEmpty } = this.props;
    const { searched } = this.state;

    if (this._header && searched && shouldSearchBarEmpty) {
      this._header.searchCancel();
    }
  }

  render() {
    const { location, loadStatus, currentFilter, onResetQueryKeyword } = this.props;
    const cancel = (isNotEmpty) => {
      if (isNotEmpty) {
        onResetQueryKeyword();
        this.setState({ searched: false });
      }
    }
    return (
      <Container>
        <PreviewHeader
          location={location}
          onSearchThunk={this._searchWrapper}
          onCancel={cancel}
          ref={ref => this._header = ref}
        />
        <PostPreviewLoader
          onScroll={this._scrollInit}
          onUnmount={this._scrollableUnmount}
          onEndReachedThreshold={SCROLLABLE_END_REACH_THRESHOLD}
          onEndReached={this._loadMore}
          refreshControl={this._renderRefreshControl()}
          ListFooterComponent={this._renderListFooter}
          removeClippedSubviews={true}
          getItemLayout={(data, index) => (
            {length: ITEM_HEIGHT, offset: ITEM_HEIGHT * index, index}
          )}
        />
        <PreviewFooter
          muted={location === ''}
          onRefresh={this._refresh}
          refreshing={loadStatus !== OK}
          popupProps={{
            onFilterThunk: this._filterWrapper,
            disabled: currentFilter
          }}
          ref={ref => this.footer = ref}
        />
      </Container>
    );
  }
}

Preview.defaultProps = {
  location: ''
}

Preview.propTypes = {
  location: PropTypes.string,
  onLoadPost: PropTypes.func.isRequired,
  onLoadMore: PropTypes.func.isRequired,
  onUpdateFilter: PropTypes.func.isRequired,
  onUpdateKeyword: PropTypes.func.isRequired,
  onUpdateQuery: PropTypes.func.isRequired,
  onResetQueryKeyword: PropTypes.func.isRequired,
  expandStatus: PropTypes.oneOf(Object.values(_expandStatus)).isRequired,
  loadStatus: PropTypes.oneOf(Object.values(_loadStatus)).isRequired,
  currentFilter: PropTypes.number.isRequired,
  shouldSearchBarEmpty: PropTypes.bool
}

const mapStateToProps = state => ({
  location: state.location.courseTitle,
  expandStatus: state.posts.subStatus,
  loadStatus: state.posts.status,
  currentFilter: state.posts.filter,
  shouldSearchBarEmpty: !state.posts.query && !state.posts.hashtag
});

const mapDispatchToProps = dispatch => ({
  onLoadPost: () => dispatch(fetchPostsSafe()),
  onLoadMore: () => dispatch(fetchExpansion()),
  onUpdateFilter: (filter) => dispatch(fetchPostsSafe(onUpdateFilter(filter))),
  onUpdateKeyword: (keyword) => dispatch(dispatch => {
    dispatch(onUpdateKeyword(keyword));
    dispatch(fetchPostsSafe());
  }),
  onUpdateQuery: (query) => dispatch(dispatch => {
    dispatch(onUpdateQuery(query));
    dispatch(fetchPostsSafe());
  }),
  onResetQueryKeyword: () => dispatch(dispatch => {
    dispatch(resetQueryKeyword());
    dispatch(fetchPostsSafe());
  })
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Preview);
