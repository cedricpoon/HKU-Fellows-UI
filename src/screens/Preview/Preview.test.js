import React from 'react';
import { shallow } from 'enzyme';
import { Preview } from './Preview';

describe('Testing Preview Screen', () => {
  const defaultProps = {
    onLoadPost: () => {}
  };

  it('renders as expected', () => {
    const wrapper = shallow(
      <Preview location='' {...defaultProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
