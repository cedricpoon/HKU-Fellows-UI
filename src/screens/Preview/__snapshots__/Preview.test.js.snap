// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Preview Screen renders as expected with last post 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Preview
    expandStatus="HALT"
    location=""
    onLoadMore={[Function]}
    onLoadPost={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withNavigation(PreviewHeader)
          location=""
        />,
        <Connect(PostPreviewLoader)
          ListFooterComponent={[Function]}
          getItemLayout={[Function]}
          onEndReached={[Function]}
          onEndReachedThreshold={0.35}
          onScroll={[Function]}
          onUnmount={[Function]}
          refreshControl={
            <RefreshControlMock
              onRefresh={[Function]}
              refreshing={false}
              tintColor="#ddd"
            />
          }
          removeClippedSubviews={true}
        />,
        <withNavigation(PreviewFooter)
          muted={true}
          onRefresh={[Function]}
        />,
      ],
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ListFooterComponent": [Function],
          "getItemLayout": [Function],
          "onEndReached": [Function],
          "onEndReachedThreshold": 0.35,
          "onScroll": [Function],
          "onUnmount": [Function],
          "refreshControl": <RefreshControlMock
            onRefresh={[Function]}
            refreshing={false}
            tintColor="#ddd"
          />,
          "removeClippedSubviews": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "muted": true,
          "onRefresh": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withNavigation(PreviewHeader)
            location=""
          />,
          <Connect(PostPreviewLoader)
            ListFooterComponent={[Function]}
            getItemLayout={[Function]}
            onEndReached={[Function]}
            onEndReachedThreshold={0.35}
            onScroll={[Function]}
            onUnmount={[Function]}
            refreshControl={
              <RefreshControlMock
                onRefresh={[Function]}
                refreshing={false}
                tintColor="#ddd"
              />
            }
            removeClippedSubviews={true}
          />,
          <withNavigation(PreviewFooter)
            muted={true}
            onRefresh={[Function]}
          />,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "location": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ListFooterComponent": [Function],
            "getItemLayout": [Function],
            "onEndReached": [Function],
            "onEndReachedThreshold": 0.35,
            "onScroll": [Function],
            "onUnmount": [Function],
            "refreshControl": <RefreshControlMock
              onRefresh={[Function]}
              refreshing={false}
              tintColor="#ddd"
            />,
            "removeClippedSubviews": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "muted": true,
            "onRefresh": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing Preview Screen renders as expected with loading more 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Preview
    expandStatus="EXPANDING"
    location=""
    onLoadMore={[Function]}
    onLoadPost={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withNavigation(PreviewHeader)
          location=""
        />,
        <Connect(PostPreviewLoader)
          ListFooterComponent={[Function]}
          getItemLayout={[Function]}
          onEndReached={[Function]}
          onEndReachedThreshold={0.35}
          onScroll={[Function]}
          onUnmount={[Function]}
          refreshControl={
            <RefreshControlMock
              onRefresh={[Function]}
              refreshing={false}
              tintColor="#ddd"
            />
          }
          removeClippedSubviews={true}
        />,
        <withNavigation(PreviewFooter)
          muted={true}
          onRefresh={[Function]}
        />,
      ],
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ListFooterComponent": [Function],
          "getItemLayout": [Function],
          "onEndReached": [Function],
          "onEndReachedThreshold": 0.35,
          "onScroll": [Function],
          "onUnmount": [Function],
          "refreshControl": <RefreshControlMock
            onRefresh={[Function]}
            refreshing={false}
            tintColor="#ddd"
          />,
          "removeClippedSubviews": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "muted": true,
          "onRefresh": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withNavigation(PreviewHeader)
            location=""
          />,
          <Connect(PostPreviewLoader)
            ListFooterComponent={[Function]}
            getItemLayout={[Function]}
            onEndReached={[Function]}
            onEndReachedThreshold={0.35}
            onScroll={[Function]}
            onUnmount={[Function]}
            refreshControl={
              <RefreshControlMock
                onRefresh={[Function]}
                refreshing={false}
                tintColor="#ddd"
              />
            }
            removeClippedSubviews={true}
          />,
          <withNavigation(PreviewFooter)
            muted={true}
            onRefresh={[Function]}
          />,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "location": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ListFooterComponent": [Function],
            "getItemLayout": [Function],
            "onEndReached": [Function],
            "onEndReachedThreshold": 0.35,
            "onScroll": [Function],
            "onUnmount": [Function],
            "refreshControl": <RefreshControlMock
              onRefresh={[Function]}
              refreshing={false}
              tintColor="#ddd"
            />,
            "removeClippedSubviews": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "muted": true,
            "onRefresh": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing Preview Screen renders as expected without load more 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Preview
    expandStatus="BLAND"
    location=""
    onLoadMore={[Function]}
    onLoadPost={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withNavigation(PreviewHeader)
          location=""
        />,
        <Connect(PostPreviewLoader)
          ListFooterComponent={[Function]}
          getItemLayout={[Function]}
          onEndReached={[Function]}
          onEndReachedThreshold={0.35}
          onScroll={[Function]}
          onUnmount={[Function]}
          refreshControl={
            <RefreshControlMock
              onRefresh={[Function]}
              refreshing={false}
              tintColor="#ddd"
            />
          }
          removeClippedSubviews={true}
        />,
        <withNavigation(PreviewFooter)
          muted={true}
          onRefresh={[Function]}
        />,
      ],
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ListFooterComponent": [Function],
          "getItemLayout": [Function],
          "onEndReached": [Function],
          "onEndReachedThreshold": 0.35,
          "onScroll": [Function],
          "onUnmount": [Function],
          "refreshControl": <RefreshControlMock
            onRefresh={[Function]}
            refreshing={false}
            tintColor="#ddd"
          />,
          "removeClippedSubviews": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "muted": true,
          "onRefresh": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withNavigation(PreviewHeader)
            location=""
          />,
          <Connect(PostPreviewLoader)
            ListFooterComponent={[Function]}
            getItemLayout={[Function]}
            onEndReached={[Function]}
            onEndReachedThreshold={0.35}
            onScroll={[Function]}
            onUnmount={[Function]}
            refreshControl={
              <RefreshControlMock
                onRefresh={[Function]}
                refreshing={false}
                tintColor="#ddd"
              />
            }
            removeClippedSubviews={true}
          />,
          <withNavigation(PreviewFooter)
            muted={true}
            onRefresh={[Function]}
          />,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "location": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ListFooterComponent": [Function],
            "getItemLayout": [Function],
            "onEndReached": [Function],
            "onEndReachedThreshold": 0.35,
            "onScroll": [Function],
            "onUnmount": [Function],
            "refreshControl": <RefreshControlMock
              onRefresh={[Function]}
              refreshing={false}
              tintColor="#ddd"
            />,
            "removeClippedSubviews": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "muted": true,
            "onRefresh": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
